test/src/stat_test.dart

```diff
@@ -1,71 +1,175 @@
+import 'dart:convert';
+
 import 'package:dcli/dcli.dart' as dcli;
-import 'package:posix/src/stat.dart';
+import 'package:posix/posix.dart' hide group;
 import 'package:test/test.dart';
 
 void main() {
-  test('stat ...', () async {
-    var tempDir = dcli.createTempDir();
-    var testFile = dcli.join(tempDir, 'test.txt');
-    dcli.touch(testFile, create: true);
-
-    print('$testFile');
-    var struct = stat(testFile);
-    var line = 'stat --format="%d,%i,%h,%u,%g,%s,%B,%b,%X,%Y,%Z,%a" $testFile '
-        .firstLine!;
-
-    var parts = line.split(',');
-    var deviceId = parts[0];
-    var inode = parts[1];
-    var nlink = parts[2];
-    var uid = parts[3];
-    var gid = parts[4];
-    var size = parts[5];
-    // var blockSize = parts[6];
-    var blocks = parts[7];
-    var lastAccess = parts[8];
-    var lastModified = parts[9];
-    var lastStatusChange = parts[10];
-    var mode = parts[11];
-    var userMode = mode[0];
-    var groupMode = mode[1];
-    var otherMode = mode[2];
-
-    // rwx
-    expect(userMode, equals('6'));
-    expect(groupMode, equals('6'));
-    expect(otherMode, equals('4'));
-
-    expect(struct.mode, isNotNull);
-
-    expect(struct.mode.isOwnerReadable, isTrue);
-    expect(struct.mode.isOwnerWritable, isTrue);
-    expect(struct.mode.isOwnerExecutable, isFalse);
-
-    expect(struct.mode.isGroupReadable, isTrue);
-    expect(struct.mode.isGroupWritable, isTrue);
-    expect(struct.mode.isGroupExecutable, isFalse);
-
-    expect(struct.mode.isOtherReadable, isTrue);
-    expect(struct.mode.isOtherWritable, isFalse);
-    expect(struct.mode.isOtherExecutable, isFalse);
-
-    expect('${struct.deviceId}', equals(deviceId));
-    expect('${struct.inode}', equals(inode));
-    expect('${struct.nlink}', equals(nlink));
-    expect('${struct.uid}', equals(uid));
-    expect('${struct.gid}', equals(gid));
-    expect('${struct.size}', equals(size));
-
-    expect('${struct.blocks}', equals(blocks));
-    expect(struct.lastAccess, equals(toDateTime(lastAccess)));
-    expect(struct.lastModified, equals(toDateTime(lastModified)));
-    expect(struct.lastStatusChange, equals(toDateTime(lastStatusChange)));
-
-    /// we don't check the blocksize as the stat command appears to always
-    /// report 512 even if the system block size is something else.
-    /// expect('${struct.blockSize}', equals(blockSize));
-  });
+    group('stat:', () {
+      late final String scratch;
+
+      setUpAll(() async {
+        scratch = _runScript('mytest', 'setup').first;
```

Comment:
I understand the intent, and try to make my own tests standalone as well. However, this set of tests needs independent manipulation of and confirmation from the filesystem and it's pretty damned hard directly in dart (which is why we need the posix package!)

Happy to use the temp directory. The scratch directory was only because it makes it much easier to confirm that the values are correct!

The bash script for setup and teardown lets us created more filesystem entities than I could find in the standard dart stuff. However, I'm happy to change if the full set of tests cases can be setup.

The bash script for pseudo (command-line) stat is more problematic. I went there because I found it very hard to get correct stat information for cross-checking. Linux and MacOS (really Darwin) stat commands are frustratingly different and hacking them around in dart is really (really) painful.

Reply:
I think dcli looks great and I'm happy to use it. Can I set file permissions (can't see chmod in the methods)? Can I create fifos? If so, I should be able to drop the mytest script.

Reply:
Is this preferable to an external script?
```
 test('file', () {
      final temp = dcli.createTempDir();
      final path = dcli.join(temp, 'test_file.txt');
      File(path).writeAsStringSync(
        '123456789\n'
        'This is a\n'
        'text file\n'
      );
      'chmod 4463 $path'.toList();
      'ln $path $temp/test_file_2.txt'.toList();
      'ln $path $temp/test_file_3.txt'.toList();
```